src/main/java/mate/academy/bookdao/BookDaoImpl.java

```diff
@@ -0,0 +1,115 @@
+package mate.academy.bookdao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.connection.ConnectionConfig;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final int INDEX_ID = 1;
+    private static final int TITLE_PART = 1;
+    private static final int PRICE_PART = 2;
+    private static final int ID_PART = 3;
+
+    @Override
+    public Book create(Book book) {
+        String addBookQuery = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionConfig.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(
+                        addBookQuery, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(TITLE_PART, book.getTitle());
+            preparedStatement.setBigDecimal(PRICE_PART, book.getPrice());
+            if (preparedStatement.executeUpdate() == 0) {

```

Comment:
This check is redundant. There is very low chance that affected rows after insert would be 0, generally you would just receive an exception.